Fix build errors with llvm: http://crbug.com/805000

These patches are already present upstream as a part of
commit ae21683d980d5fe9e39bd0193827ea3604256eb9
Author: Jon Olav Hauglid <jon.hauglid@oracle.com>
Date:   2013-02-28

    Bug#16265300: Fix new compilation warnings from XCode 4.6

    This patch fixes three new compilation warnings generated by Clang.
    It also adds -Wall flag for Clang and disables -Wnull-conversion
    and -Wunused-private-field.

https://github.com/mysql/mysql-server/commit/ae21683d980d5fe9e39bd0193827ea3604256eb9#diff-e2295af0c82aeb3e57770407501024b4
https://github.com/mysql/mysql-server/commit/ae21683d980d5fe9e39bd0193827ea3604256eb9#diff-3ff5b68f3c27cf9b534782e7d60170c1

--- mysql/client/mysql.cc
+++ mysql/client/mysql.cc
@@ -2831,7 +2831,7 @@ You can turn off this feature to get a quicker startup with -A\n\n");
         mysql_free_result(fields);
         break;
       }
-      field_names[i][num_fields*2]= '\0';
+      field_names[i][num_fields*2]= NULL;
       j=0;
       while ((sql_field=mysql_fetch_field(fields)))
       {
--- mysql/sql/sql_partition.cc
+++ mysql/sql/sql_partition.cc
@@ -282,7 +282,7 @@ bool partition_default_handling(TABLE *table, partition_info *part_info,
     }
   }
   part_info->set_up_defaults_for_partitioning(table->file,
-                                              (ulonglong)0, (uint)0);
+                                              NULL, 0U);
   DBUG_RETURN(FALSE);
 }
 
